themes/phudj/phudj.theme.bash

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env bash
+
+  #####################################
+ #  https://github.com/phreakocious  #
+#####################################
+
+#
+# phudj is a theme for bash-it -- https://github.com/Bash-it/bash-it
+# 
+
+# ★ ➤ ▶ ► ‣ ✱ ✽ ❖ ✸ ✹ ◼■ ●○-°✪ ⧫⬧♦⬥ ⁉ ‼ ✗ ✘ ❓❔ ⁈ ※ ⁂ ⁑ ⧖ ⧗ ☆ ⚠
+# ❗⚡ ✅ ❌ ✨❓ 🌀 🌈 🔥 🥓 🌞 💫 💥 💣 💡🌟⛔ ⬤ ⚫⭕ 🔴 🔵 🔶 🔷 🔸 🔹 🔺 🔻 🚫 🚀 ☄ 💢 
+
+      prompt_user=▶
+     prompt_clean=✔
+     prompt_dirty=✗
+     prompt_ahead=↑
+    prompt_behind=↓
+   prompt_diverge=↕
+     prompt_stash=✱
+
+      color_user="$bold_green"
+      color_host="$green"
+      color_time="\[$(echo_color rgb 265 265 265)\]"
+        color_at="$color_time"
+      color_path="$cyan"
+     color_clean="$bold_green"
+     color_dirty="\[$(echo_color rgb 255 95 0)\]"
+     color_ahead="$bold_yellow"
+    color_behind="$yellow"
+   color_diverge="$bold_white"
+    color_branch="\[$(echo_color rgb 0 255 95)\]"
+   color_success="$bold_green"
+   color_failure="$bold_red"
+     color_stash="\[$(echo_color rgb 223 15 31)\]"
+
+prompt_user_root=★
+ color_user_root="$bold_red"
+ color_host_root="$red"
+
+###
+
+
+[[ $EUID -eq 0 ]] && color_host=${color_host_root} && color_user=${color_user_root} && prompt_user="$prompt_user_root"
+
+# seq/jot are not always available, so eval+echo+brace expansion it shall be
+repeat_char() { char=$1; reps=$2; printf "${char}%.0s" $(eval echo {1..$reps}) ; }
+
+parse_git() {
+  git_status="$(git status --show-stash 2>/dev/null)" || return
```

Comment:
Parsing `git status` output is generally discouraged, since the output may change between Git versions, and the text output can differ based on the configured language of the user. The patterns you're using below will only work for English language settings, for example.

In general, you should use `git status --porcelain` for things like this... Or even better use one of the existing Bash-it functions for this.

Reply:
Understood.  From what I could see, `porcelain` does not include information about stashes (despite the `--show-stashes` option.  Running `git status` is expensive, especially on remote file systems, so I'm avoiding running it multiple times for each prompt.  Using the built in `git_prompt_vars` is considerably slower.

Reply:
I see your point on the stash information.

But on the other hand your parsing relies on English language output. This will not work for people that run their shell in a different language. Your parsing code is going to fail with other languages...